lib/fastly_nsq/message_queue_listener.rb

```diff
@@ -0,0 +1,27 @@
+module MessageQueue
+  class Listener
+    def initialize(topic:, channel:)
+      @topic = topic
+      @channel = channel
+    end
+
+    def start
+      loop do
+        process_next_message
+      end
+    end
+
+    def process_next_message
+      message = consumer.pop
+      MessageProcessor.new(message).start
```

Comment:
As I said in the Documentation comment, we should call out to something else here... `process`